tests/baselines/reference/unicodeEscapesInNames02(target=es5).js

```diff
@@ -0,0 +1,95 @@
+//// [tests/cases/compiler/unicodeEscapesInNames02.ts] ////
+
+//// [extendedEscapesForAstralsInVarsAndClasses.ts]
+// Example from https://mathiasbynens.be/notes/javascript-identifiers-es6
+// Astral characters should be accepted in ES2015
+
+// U+102A7 CARIAN LETTER A2
+var 𐊧: string;
+var \u{102A7}: string;
+
+if (Math.random()) {
+    𐊧 = "hello";
+}
+else {
+    \u{102A7} = "hallo";
+}
+
+class Foo {
+    \u{102A7}: string;
+    constructor() {
+        this.\u{102A7} = " world";
+    }
+    methodA() {
+        return this.𐊧;
+    }
+}
+
+export var _𐊧 = new Foo().\u{102A7} + new Foo().methodA();
+
+_\u{102A7} += "!";
+
+//// [astralAsSurrogatePair.ts]
+import { _𐊧 as \uD800\uDEA7 } from "./extendedEscapesForAstralsInVarsAndClasses.js";
+
+
+//// [extendedEscapesForAstralsInVarsAndClasses.js]
+// Example from https://mathiasbynens.be/notes/javascript-identifiers-es6
+// Astral characters should be accepted in ES2015
+// U+102A7 CARIAN LETTER A2
+var string;

```

Comment:
This looks off?

Reply:
But maybe it's expected when there are errors?

Reply:
The ES5 emit is completely broken because the parsing fails badly. If we allow ES2015+ identifiers to be parsed correctly in an ES5 target and issue an error after the fact, then this file will proably be fixed-ish

Reply:
Would emitting an escape sequence be an option?

Reply:
It's not quite that simple because you can't "downlevel" every identifier to ES5 if it contains non-ES5 characters. Doing so would imply a stable name-mangling scheme for properties in the following case:

```ts
export class C {
    // if we downlevel this to ES5, how do library consumers access this property?
    \u{102A7} = 42;
}
```